### PR TITLE
Add helm chart files (first step in helm support)

### PR DESCRIPTION
**Pull request by kuba-wolf**
Tested by running these commands:
```
helm install test charts/aws-file-cache-csi-driver --namespace kube-system

kubectl annotate serviceaccount -n kube-system file-cache-csi-controller-sa \
 eks.amazonaws.com/role-arn=arn:aws:iam::651262257939:role/HelmFileCacheRole --overwrite=true

# delete the two csi controller pods so new ones are spun up and the role is applied

kubectl delete -f examples/kubernetes/dynamic_provisioning/specs/claim.yaml
kubectl delete -f examples/kubernetes/dynamic_provisioning/specs/storageclass.yaml
```

Also made multiple code QOL fixes suggested in this PR: https://github.com/kubernetes-sigs/aws-file-cache-csi-driver/pull/5

Will follow up this PR with 1) adding gh-pages branch and 2) adding github workflow to automatically push to the gh-pages branch

### CODE REVIEW

File: examples/kubernetes/dynamic_provisioning/specs/storageclass.yaml [hunk 1]
```diff
@@ -5,13 +5,7 @@ metadata:
...
   subnetId: subnet-074ea3b7386e4802b
-  securityGroupIds: sg-046ad64768e4cf756,sg-02c92284b2c124cb2
+  securityGroupIds: sg-046ad64768e4cf756
   dataRepositoryAssociations: "FileCachePath=/ns1/,DataRepositoryPath=nfs://10.0.92.69/,NFS={Version=NFS3},DataRepositorySubdirectories=[subdir1,subdir2,subdir3]"
-  fileCacheType: "LUSTRE"
-  fileCacheTypeVersion: "2.12"
-  weeklyMaintenanceStartTime: "7:00:00"
-  LustreConfiguration: "DeploymentType=CACHE_1,PerUnitStorageThroughput=1000,MetadataConfiguration={StorageCapacity=2400}"
-  copyTagsToDataRepositoryAssociations: "true"
-  extraTags: "Tag1=Value1,Tag2=Value2"
```
Comment (PatrickGhadban) [inline examples/kubernetes/dynamic_provisioning/specs/storageclass.yaml:15]: Could we add these back in the follow up CRs you make?

File: pkg/util/util.go [hunk 2]
```diff
@@ -101,3 +101,14 @@ func SplitUnnestedCommas(input string) []string {
 	slices = append(slices, input[lastComma:])
 	return slices
 }
+
+func MapValues(input []string) map[string]string {
```
Comment (PatrickGhadban) [inline pkg/util/util.go:105]: Nice!

**Review by PatrickGhadban — APPROVED** [review]

File: examples/kubernetes/dynamic_provisioning/specs/storageclass.yaml (hunk 1 above)
Comment (kuba-wolf) [reply examples/kubernetes/dynamic_provisioning/specs/storageclass.yaml:15]: Yup

**Pull request merged**